.github/workflows/cache.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow caches crate dependencies and build artifacts for tools (except 'test-tools' since we don't use them in build workflows).
+# The cache is only usable by workflows started from pull requests against the develop branch.
+name: CacheDepsAndTools
+on:
+  push:
+    branches: [develop]
+    paths:
+      - '.github/**'
+      - 'sources/Cargo.lock'
+      - 'tools/buildsys/**'
+      - 'tools/pubsys*/**'
+      - '!tools/pubsys/policies/**'
+      - '!tools/pubsys/**.example'
+      - '!tools/pubsys/**.template'
+      - 'tools/Cargo.lock'
+jobs:
+  cache:
+    runs-on:
+      group: bottlerocket
+      labels: bottlerocket_ubuntu-latest_16-core
+    continue-on-error: true
+    steps:
+      - run: |
+          echo "OS_ARCH=`uname -m`" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+      # Cache `cargo-make`, `cargo-cache`, `cargo-sweep`
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}
+      # Cache first-party code dependencies
+      - uses: actions/cache@v3
+        with:
+          path: |
+            .cargo
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('sources/Cargo.lock') }}
+      # Cache 'tools/' dependencies and build artifacts
+      - uses: actions/cache@v3
+        with:
+          path: |
+            tools/bin
+            tools/.crates.toml
+            tools/.crates2.json
+            tools/target
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('tools/Cargo.lock') }}
+      - run: rustup default 1.64.0
+      - run: cargo install --locked --version 0.36.0 cargo-make
+      - run: cargo install --locked --version 0.8.3 --no-default-features --features ci-autoclean cargo-cache
```

Comment:
I wonder if we really need to pin the versions for `ci-autoclean`, `cargo-cache`, and `cargo-sweep`. I worry that we'll either forget to bump them, or worse bump them in some places but not in others. I guess the same could be said for `cargo-make` as well.

Reply:
I'm more afraid of breaking changes bring introduced. We've been burned by that with `cargo-make` more than once. This freezes with versions of the tools we know that will work.

Reply:
I think `cargo-make` has a higher chance of that occurring since it is dealing with the build itself. At least with `cargo-cache` I've never really ran into a regression. It seems like something occurring in these housekeeping packages is less likely, and would be easier to tackle if something did occur.

Reply:
I've been burned by new breaking changes too, so I'm usually more comfortable explicitly pinning like this and making sure upgrades are a conscious decision.

If we want to try to keep up to date, dependabot can be configured to propose GitHub Action updates for these too.

Reply:
Oh, I really like the idea of dependabot doing it. That way we can get a heads up of when we do get a breaking change, but it wouldn't break our PRs across the board. It would also make sure all workflows are getting the same bumps.

Reply:
> I'm more afraid of breaking changes bring introduced.

> I've been burned by new breaking changes too

+1 to that - Also in favor of pinning versions of these tools. And yes! I also like the idea of getting dependabot to do it. However, speaking of, where in our project do we declare those things as dependencies? The only place I see it [defined as a dependency is in our `BUILDING.md` file.](https://github.com/bottlerocket-os/bottlerocket/blob/develop/BUILDING.md#rust) In order for depndabot to manage it, we'd have to codify it in some `Cargo.toml` file or something.

Reply:
Dependabot understands these things separately from Rust or Go code: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot

We would just need to add that configuration file. But we might have dependabot completely disabled on this main repo right now, so we would probably need someone with admin rights to turn that one once we have the config file in place.

Reply:
Figured I might as well just throw that up there: https://github.com/bottlerocket-os/bottlerocket/pull/2655

Reply:
We disabled dependabot PRs for the entire repository since it was introducing way too much PR noise for rust crate bumps. We know to do that before every release so there was limited-to-no value with the dependabot PRs. We still have security notification enabled which we do rely on from time to time.

`cargo-make` introduced breaking changes between patch version before so limiting the notifications to just minor version bumps doesn't really help and just introduces PR noise. And honestly, since these `cargo-*` tools are extremely limited in scope (as is with the workflow itself), it's very hard for any kind of security vulnerability with them to affect us in any meaningful way (e.g. we don't use the build artifacts etc). At worse, `cargo-make` somehow would give us a false positive or caches fill up because `cargo-sweep` doesn't work (but we know it does right now)

Reply:
> We disabled dependabot PRs for the entire repository since it was introducing way too much PR noise for rust crate bumps.

Right, but if the scope is limited to the couple crates of actions it shouldn't be too bad. We definitely don't want to turn dependabot on for the whole repo again.